tests/Unit/ParallelAlgorithms/SurfaceFinder/Test_SurfaceFinder.cpp

```diff
@@ -0,0 +1,170 @@
+  const auto inertial_contour = element_map(result);
+  const auto radius_values = get(magnitude(inertial_contour));
+
+  Approx custom_approx = Approx::custom().epsilon(1.e-3).scale(1.0);
+  CHECK(radius_zeta[0].has_value() == false);
+  CHECK(radius_zeta[1].has_value() == true);
+  CHECK(radius_values == custom_approx(sqrt(-25. * log(0.25)) - 1.));
+  CHECK(radius_zeta[2].has_value() == false);
+}
+
+void test_radius_contour(
+    const tnsr::I<DataVector, 3, Frame::Inertial>& inertial_coords,
+    const Mesh<3>& mesh,
+    const tnsr::I<DataVector, 2, Frame::ElementLogical>& ray_directions,
+    const ElementMap<3, Frame::Inertial>& element_map) {
+  INFO("Find certain radius contour.");
+
+  const auto data = magnitude(inertial_coords);
+  const double target = 4.0;
+
+  const auto radius_zeta =
+      SurfaceFinder::find_radial_surface(data, target, mesh, ray_directions);
+  const auto result = result_helper(ray_directions, radius_zeta);
+  const auto inertial_contour = element_map(result);
+  const auto radius_values = get(magnitude(inertial_contour));
+
+  Approx custom_approx = Approx::custom().epsilon(1.e-3).scale(1.0);
+  for (size_t i = 0; i < radius_zeta.size(); i++) {
+    CHECK(radius_zeta[i].has_value());
+    CHECK(radius_values[i] == custom_approx(target));
+  }
+}
+
+void test_strahlkorper_input(
+    const tnsr::I<DataVector, 3, Frame::Inertial>& inertial_coords,
+    const Mesh<3>& mesh, const Domain<3>& domain, const ElementId<3>& id,
+    const ElementMap<3, Frame::Inertial>& element_map) {
+  INFO(
+      "Take the collocation points from a Strahlkorper, find a particular "
+      "radius, and compare to the physical radius from the Strahlkorper.");
+
+  const double target = 4.3;
+  const auto data = magnitude(inertial_coords);
+
+  const Strahlkorper<Frame::Inertial> strahlkorper{
+      5, 5, 4.5, std::array<double, 3>{0., 0., 0.}};
+  const auto& ylm = strahlkorper.ylm_spherepack();
+  const auto& [theta, phi] = ylm.theta_phi_points();
+  const auto x = target * sin(theta) * cos(phi);
+  const auto y = target * sin(theta) * sin(phi);
+  const auto z = target * cos(theta);
+  const tnsr::I<DataVector, 3, Frame::Inertial> points{{{x, y, z}}};
+  const std::vector<ElementId<3>> id_vector{id};
+  const auto block_coords = block_logical_coordinates(domain, points);
+  auto element_coords = element_logical_coordinates(id_vector, block_coords);
+  const tnsr::I<DataVector, 2, Frame::ElementLogical> ray_directions{
+      {{get<0>(element_coords.at(id).element_logical_coords),
+        get<1>(element_coords.at(id).element_logical_coords)}}};
+
+  const auto radius_zeta =
+      SurfaceFinder::find_radial_surface(data, target, mesh, ray_directions);
+  const auto result = result_helper(ray_directions, radius_zeta);
+  const auto inertial_contour = element_map(result);
+  const auto radius_values = get(magnitude(inertial_contour));
+
+  Approx custom_approx = Approx::custom().epsilon(1.e-3).scale(1.0);
```

Comment:
Can you tighten the tolerances in the tests if you use 12 points in all dimensions or 1 level of h refinement? With h refinement you just have to pick a suitable element that covers the surface, like `ElementId<3>{0, {{{1, 0}, {1, 0}, {1, 1}}}` or something like that. That's Block 0 (look in Sphere.cpp to understand which direction wedge that is), refinement level 1 in all directions, and the first element of two in angular directions and second element of two radially.

Reply:
I have tightened the tolerances by adding more points. To do so with h refinement would require rewriting one of the tests (the "bulging surface"), however if these still aren't tight enough I am happy to do so. :)